pkgs/jni/test/jarray_test.dart

```diff
@@ -48,24 +60,44 @@ void run({required TestRunnerCallback testRunner}) {
     using((arena) {
       final array = JArray(jchar.type, 3)..releasedBy(arena);
       expect(array.length, 3);
-      array[0] = 'ح';
-      array[1] = '2';
-      array[2] = '3';
-      expect(array[0], 'ح');
-      expect(array[1], '2');
-      expect(array[2], '3');
-      array.setRange(0, 3, ['4', '5', '6', '7'], 1);
-      expect(array[0], '5');
-      expect(array[1], '6');
-      expect(array[2], '7');
+      array[0] = 'ح'.codeUnitAt(0);
```

Comment:
This happens to work because `  print('ح'.codeUnits);` shows it's a single code unit.
Please write the test in a way that deals with having multiple code units per character.
Try `print('🧑🏿‍🤝‍🧑🏿'.codeUnits);`

Reply:
I don't particularly care about the number of `codeUnits` here, it simply was easier to change this test by adding `codeUnitAt(0)` everywhere, to convert from the **incorrect** character array we had before, to an Int16 array.

I can change the test to just use numbers, this way we don't have to think about Dart's `codeUnits` at all.

Reply:
Yeah, using ints would be better.

I still think it's worth updating doc comments on char arrays to point out one would store code units in there so that users don't expect the index operator to be a character in a string. (Changing it to int already makes it much better than a single char string!)